opencensus-shim/src/main/java/io/opentelemetry/opencensusshim/OpenTelemetryNoRecordEventsSpanImpl.java

```diff
@@ -0,0 +1,68 @@
+package io.opentelemetry.opencensusshim;
+
+import com.google.common.base.Preconditions;
+import io.opencensus.trace.Annotation;
+import io.opencensus.trace.AttributeValue;
+import io.opencensus.trace.EndSpanOptions;
+import io.opencensus.trace.Link;
+import io.opencensus.trace.MessageEvent;
+import io.opencensus.trace.Span;
+import io.opencensus.trace.SpanContext;
+import io.opencensus.trace.Status;
+import java.util.EnumSet;
+import java.util.Map;
+
+public class OpenTelemetryNoRecordEventsSpanImpl extends Span {
+  private static final EnumSet<Options> NOT_RECORD_EVENTS_SPAN_OPTIONS =
+      EnumSet.noneOf(Options.class);
+
+  static OpenTelemetryNoRecordEventsSpanImpl create(SpanContext context) {
+    return new OpenTelemetryNoRecordEventsSpanImpl(context);
+  }
+
+  @Override
+  public void addAnnotation(String description, Map<String, AttributeValue> attributes) {
+    Preconditions.checkNotNull(description, "description");
+    Preconditions.checkNotNull(attributes, "attribute");
+  }
+
+  @Override
+  public void addAnnotation(Annotation annotation) {
+    Preconditions.checkNotNull(annotation, "annotation");
+  }
+
+  @Override
+  public void putAttribute(String key, AttributeValue value) {
+    Preconditions.checkNotNull(key, "key");
+    Preconditions.checkNotNull(value, "value");
```

Comment:
nit: Do we even need to perform these checks if the implementation is a no-op?

Reply:
This seems like less of a nit, and more of a problem. If it throws exceptions on bad inputs, it's not a no-op, I think.

Reply:
I actually just copied all of this from https://github.com/census-instrumentation/opencensus-java/blob/f0e78ba5a10165cce2b63809bfdfe3e34b1eb456/impl_core/src/main/java/io/opencensus/implcore/trace/NoRecordEventsSpanImpl.java
Not sure why there are checks in the OC implementation, happy to remove the checks here though if it makes more sense

Reply:
If OC users are fine with that behavior, then it's cool. We've taken a different approach in OTel, but this is fine if it's what users are used to.